### PR TITLE
[5.7] Allow for explicit text parameters on Authorize middleware

### PR DESCRIPTION
**Pull request by emiliopedrollo**
Pull Request #25763 broke some non documented functionality (explained here #26422) related to the Authorize middleware.

The previous implementation would allow to set an Authorize for a Gate with an optional parameter. That could be useful when the 'can' middleware is attached directly into the controller method (instead of a route) and thus act on multiple routes that access such method.

The current (master) implementation sends a string with the value of the Authorize argument if it's not a route parameter. This is behave ambiguous as any argument on the middleware can mean either a parameter of the request named after it or it's own name depending of the conditions a layer of abstraction up.

This PR solves this issue while still allowing the Authorize middleware to send string values to Gates by explicitly defining them as string using single or double quoted values on it's arguments, thus removing the ambiguity:
````php
    // this should only send and route parameter to the gate or null if not available
    Route::middleware('can:ability,model');
````
````php
    // This will send a string to the gate
    Route::middleware('can:ability,"some text"');
````

TL;DR; This reverts the default bahaviour to prior PR #25763, while still allowing string parameters.

### CODE REVIEW

**Comment by taylorotwell** [comment]
I'm not sure I plan to change this behavior at this time.
